hecat/processors/github_metadata.py

```diff
@@ -76,41 +67,183 @@ def write_software_yaml(step, software):
+                        github_projects.append(software)
+                    else:
+                        logging.debug('all metadata already present, skipping %s', software['source_code_url'])
+                # If key is not present, update all metadata
                 else:
-                    logging.debug('all metadata already present, skipping %s', github_url)
+                    github_projects.append(software)
+    # Get the metadata for the GitHub repositories
+    GITHUB_GRAPHQL_API = "https://api.github.com/graphql"
+    headers = {"Authorization": f"Bearer {GITHUB_TOKEN}"}
+
+    # Get the URLs of the queued repositories
+    github_urls = [software['source_code_url'] for software in github_projects]
+    repos = [re.sub('https://github.com/', '', url) for url in github_urls]
+
+    # This limit was tested with a personal access token, batch_size = 30, timeout = 60 (new default if not provided in a config); Worked fine for the full repo (enable debug messages to see usage of API Rate limit stats)
+    # Split repo list into batches of batch_size
+    if 'batch_size' in step['module_options']:
+        batch_size = step['module_options']['batch_size']
+    else:
+        # Default batch_size of repos if not specified in config file
+        batch_size = 30
+    batches = [repos[i * batch_size:(i + 1) * batch_size] for i in range((len(repos) + batch_size - 1) // batch_size )]
+
+    counter = 0
+    for batch in batches:
+        counter += 1
+        logging.info(f"Processing batch {counter}/{len(batches)}")
+        repos_query = " ".join([f"repo:{repo}" for repo in batch])
+        now = datetime.now()
+        year_month = now.strftime("%Y-%m")
+        # TODO: More accurate lookup would be for last month, but would mean we lag 1 month behind
+        #dateMonth = (now.month - 2) % 12 + 1
+        #dateYear = now.year - 1 if dateMonth == 12 else now.year
+        #year_last_month = datetime.date(dateYear, dateMonth, 1).strftime("%Y-%m")
+
+        query = f"""
+{{
+    search(
+        type: REPOSITORY
+        query: "{repos_query}"
+        first: {batch_size}
+    ) {{
+    repos: edges {{
+        repo: node {{
+        ... on Repository {{
+            url
+            stargazerCount
+            isArchived
+            releases(first: 1) {{
+                edges {{
+                    node {{
+                        tagName
+                        publishedAt
+                    }}
+                }}
+            }}
+            defaultBranchRef {{
+                target {{
+                    ... on Commit {{
+                        committedDate
+                        history(since: "{year_month}-01T00:00:00", until: "{year_month}-31T23:59:59") {{
+                            totalCount
+                        }}
+                    }}
+                }}
+            }}
+        }}
+        }}
+    }}
+    }}
+}}
+        """
+        res_header = None
```

Comment:
```suggestion
        graphql_response_header = None
```
descriptive variable names (please also replace below)